server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/toolServer.ts

```diff
@@ -73,28 +82,89 @@ export const LspToolsServer: Server = ({ workspace, logging, lsp, agent }) => {
-    lsp.onInitialized(async () => {
-        // todo: move to constants
-        var workspaceFolders = workspace.getAllWorkspaceFolders()
-        const wsUris = workspaceFolders?.map(f => f.uri) ?? []
-        const wsConfigPaths = wsUris.map(uri => `${uri}/.amazonq/mcp.json`)
-        const globalConfigPath = `${workspace.fs.getUserHomeDir()}/.aws/amazonq/mcp.json`
-        const allPaths = [...wsConfigPaths, globalConfigPath]
-
-        const mgr = await McpManager.init(allPaths, { logging, workspace, lsp })
-
-        for (const def of mgr.getAllTools()) {
-            const baseSpec = def
-            const namespaced = `${def.serverName}_${def.toolName}`
+export const McpToolsServer: Server = ({ credentialsProvider, workspace, logging, lsp, agent, telemetry, runtime }) => {
+    const registered: Record<string, string[]> = {}
+
+    const allNamespacedTools = new Set<string>()
+
+    function registerServerTools(server: string, defs: McpToolDefinition[]) {
+        // 1) remove old tools
+        for (const name of registered[server] ?? []) {
+            agent.removeTool(name)
+        }
+        registered[server] = []
+
+        // 2) add new enabled tools
+        for (const def of defs) {
+            const namespaced = createNamespacedToolName(
+                def.serverName,
+                def.toolName,
+                allNamespacedTools,
+                McpManager.instance.getToolNameMapping()
+            )
             const tool = new McpTool({ logging, workspace, lsp }, def)
 
+            // Add explanation field to input schema
+            const inputSchemaWithExplanation = {
+                ...def.inputSchema,
+                properties: {
+                    ...def.inputSchema.properties,
+                    explanation: {
+                        type: 'string',
+                        description:
+                            'One sentence explanation as to why this tool is being used, and how it contributes to the goal.',
+                    },
+                },
+            }
+
             agent.addTool(
-                { name: namespaced, description: baseSpec.description, inputSchema: baseSpec.inputSchema },
-                (input: any) => tool.invoke(input)
+                { name: namespaced, description: def.description, inputSchema: inputSchemaWithExplanation },
+                input => tool.invoke(input)
             )
-            logging.info(`MCP: registered tool ${namespaced}`)
+            registered[server].push(namespaced)
+            logging.info(`MCP: registered tool ${namespaced} (original: ${def.serverName}___${def.toolName})`)
+        }
+    }
+
+    lsp.onInitialized(async () => {
+        if (!enabledMCP(lsp.getClientInitializeParams())) {
+            logging.warn('MCP is currently not supported')
+            return
         }
+
+        const wsUris = lsp.getClientInitializeParams()?.workspaceFolders?.map(f => f.uri) ?? []
```

Comment:
This also needs to follow #1353 - please check the entire branch for any direct workspaceFolders lookups, I may have missed some